WoloxAndroidBootstrp/app/src/main/java/ar/com/wolox/android/example/ui/news/NewsFragment.kt

```diff
@@ -1,18 +1,72 @@
 package ar.com.wolox.android.example.ui.news
 
+import android.os.Bundle
+import androidx.core.content.ContextCompat
+import androidx.core.os.bundleOf
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
 import ar.com.wolox.android.R
 import ar.com.wolox.android.databinding.FragmentNewsBinding
+import ar.com.wolox.android.example.model.News
+import ar.com.wolox.android.example.utils.Extras
+import ar.com.wolox.android.example.utils.RequestCode
 import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import ar.com.wolox.wolmo.core.util.ToastFactory
 import javax.inject.Inject
 
-class NewsFragment @Inject constructor() : WolmoFragment<FragmentNewsBinding, NewsPresenter>(), NewsView {
+class NewsFragment @Inject constructor() : WolmoFragment<FragmentNewsBinding, NewsPresenter>(), NewsView, SwipeRefreshLayout.OnRefreshListener {
+
+    @Inject
+    internal lateinit var toastFactory: ToastFactory
 
     override fun layout() = R.layout.fragment_news
 
+    override fun handleArguments(arguments: Bundle?) = arguments?.containsKey(Extras.News.KEY_NAME)
+
     override fun init() {
+        presenter.onInit(requireArgument(Extras.News.KEY_NAME))
+        with(binding) {
+            swipeRefreshNewsFragment.setOnRefreshListener(this@NewsFragment)
+        }
+    }
+
+    override fun setListeners() {
+        binding.ButtonNewsCardFav.setOnClickListener {
+            it.background = ContextCompat.getDrawable(requireContext(), R.drawable.ic_like_on)
+            presenter.setLikeRequest()
```

Comment:
We usually call presenter functions as events. Try with `presenter.onLikeButtonClicked()`, for example.